hawkmoth/parser.py

```diff
@@ -389,8 +401,15 @@ def _type_fixup(cursor):
         name = cursor.spelling + dims
 
     inheritance = _get_inheritance(cursor)
+    scopedenum_type = _get_scopedenum_type(cursor)
 
-    return ttype, f'{name}{inheritance if inheritance else ""}'
+    colon_suffix = ''
+    if inheritance:
+        colon_suffix = inheritance
+    if scopedenum_type:
+        colon_suffix = scopedenum_type
+
+    return ttype, f'{name}{colon_suffix}'
```

Comment:
Same as previous patch, is it necessary to handle scoped enum type within `_type_fixup()`, or could it just be a separate thing for scoped enums?

Reply:
Yeah, like before, I agree with the sentiment despite a few common lines of code. I think we should be able to at least split new type declarations from variable / function argument declarations.